rest_framework_simplejwt/models.py

```diff
@@ -7,10 +7,17 @@
 from .settings import api_settings
 
 if TYPE_CHECKING:
+    from django.contrib.auth.models import AbstractBaseUser
+
     from .tokens import Token
 
+    TokenUserBase: TypeAlias = AbstractBaseUser
+
+else:
+    TokenUserBase = object
+
 
-class TokenUser:
+class TokenUser(TokenUserBase):
```

Comment:
why is this being introduced?

Is there a reason why `TokenUser` should be a subclass of `AbstractBaseUser`?

Reply:
I have a feeling it's because, unlike Typescript, there isn't interface overlapping in Python, so the author was trying to resolve what happens when you receive a "user" type.

They made a change here that explains it a little:
```suggestion
- AuthUser = TypeVar("AuthUser", AbstractBaseUser, TokenUser)
+ TokenTypeVar = TypeVar("TokenTypeVar", bound=Token)
```